src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1470,9 +1497,36 @@ public void ProcessFileChange(string projectSystemFilePath, string workspaceFile
 
                 ClearAndZeroCapacity(_documentsRemovedInBatch);
 
+                // Update project's order of documents.
+                solution = this.UpdateProjectDocumentsOrderForBatch(solution, _project.Id);
+
                 return solution;
             }
 
+            internal Solution UpdateProjectDocumentsOrderForBatch(Solution solution, ProjectId projectId)
+            {
+                // The action of ordering documents was not included in the batch. Simply return the solution.
+                if (_multipleOrderedFilePathsInBatch.Count == 0)
+                {
+                    return solution;
+                }
+
+                var orderedFilePaths = _multipleOrderedFilePathsInBatch.Last();
+
+                ClearAndZeroCapacity(_multipleOrderedFilePathsInBatch);
+
+                // We only care about the latest file paths.
```

Comment:
nit: seems this comment should be moved to the assignment declaration above.

Reply:
This has been re-done.